context/db.go

```diff
@@ -68,15 +69,26 @@ func SetupAmfCollection() {
 		factory.AmfConfig.Configuration.AmfDBName = "sdcore_amf"
 	}
 	if (factory.AmfConfig.Configuration.Mongodb == nil) ||
-		(factory.AmfConfig.Configuration.Mongodb.Url == "") {
-		mongoDbUrl = "mongodb://mongodb:27017"
+		factory.AmfConfig.Configuration.Mongodb.Url == "" {
+		mongoDbUrl = "mongodb://mongodb-arbiter-headless"
+	} else {
+		mongoDbUrl = factory.AmfConfig.Configuration.Mongodb.Url
 	}
 
+	logger.ContextLog.Infof("MondbName: %v, Url: %v", factory.AmfConfig.Configuration.AmfDBName, mongoDbUrl)
+
 	if Namespace != "" {
 		AmfUeDataColl = Namespace + "." + AmfUeDataColl
 	}
-
-	MongoDBLibrary.SetMongoDB(factory.AmfConfig.Configuration.AmfDBName, mongoDbUrl)
+	for {
+		MongoDBLibrary.SetMongoDB(factory.AmfConfig.Configuration.AmfDBName, mongoDbUrl)
+		if MongoDBLibrary.Client == nil {
```

Comment:
we have a new function which returns the client to you I think. Its better to use that than directly access variable in library.